webapi/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -339,30 +383,41 @@ public async Task<string> ExtractAudienceAsync(SKContext context, CancellationTo
     /// <param name="chatId">The chat ID</param>
     /// <param name="userId">The user ID</param>
     /// <param name="chatContext">The SKContext.</param>
+    /// <param name="userMessage">ChatMessage object representing new user message.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>The created chat message containing the model-generated response.</returns>
-    private async Task<ChatMessage> GetChatResponseAsync(string chatId, string userId, SKContext chatContext, CancellationToken cancellationToken)
+    private async Task<ChatMessage> GetChatResponseAsync(string chatId, string userId, SKContext chatContext, ChatMessage userMessage, CancellationToken cancellationToken)
     {
+        // Render system instruction components
+        var promptRenderer = new PromptTemplateEngine();
+        var systemInstructions = await promptRenderer.RenderAsync(
+            this._promptOptions.SystemPersona,
+            chatContext,
+            cancellationToken);
```

Comment:
can we add a status update for this async call?
```suggestion
        await this.UpdateBotResponseStatusOnClient(chatId, "Rendering prompt template", cancellationToken);
        var promptRenderer = new PromptTemplateEngine();
        var systemInstructions = await promptRenderer.RenderAsync(
            this._promptOptions.SystemPersona,
            chatContext,
            cancellationToken);
```

Reply:
Added an update with status "Initializing prompt".